### ISSUE TITLE
csm_allocation_query -j cannot return multiple records

### ISSUE THREAD

**Issue by awlauria** [IBM/CAST#275]
**Describe the bug**
In cases where there are multiple job ids, CSM returns a generic error like the following:

"Multiple jobs of this id exists"

instead of returning all matching ids, which I would expect.

**To Reproduce**
1. Get 2 bsubs with the same jobid
2. run csm_allocation_query -j on that ID

**Expected behavior**
All records with that ID would be returned.

**Environment (please complete the following information):**
Reproduced on WSC.

**Issue Source:**

There are cases where we can "lose" the allocation_id in our environment (such as exiting from the launch node and sshing back in). The easiest way to get the ID back is to query csm using the job id. But, on the rare occasions described here, this is not useful. Even if it returns all such ID's, if sorted from "oldest to newest", it would be trivial to find my allocation.

**Comment by fpizzano** [comment]
Few more details:

```
# /opt/ibm/csm/bin/csm_allocation_query -j 293757
[csmapi][warning]	/home/ppsbld/workspace/PUBLIC_CAST_V1.4.x_ppc64LE_RH7.5_ProdBuild/csmi/src/common/src/csmi_common_utils.c-147: the Error Flag Set
[csmapi][error]	csmi_sendrecv_cmd failed: 25 - csm_allocation_query[114019530]; Too many allocations matching the query were found, something has gone wrong.
/opt/ibm/csm/bin/csm_allocation_query FAILED: errcode: 25 errmsg: csm_allocation_query[114019530]; Too many allocations matching the query were found, something has gone wrong.
```

```
 allocation_id | primary_job_id | launch_node_name | num_nodes | user_name |         begin_time         |          end_time
---------------+----------------+------------------+-----------+-----------+----------------------------+----------------------------
        207977 |         293757 | batch3           |        20 | lot       | 2019-03-11 16:03:30.561825 | 2019-03-11 16:05:03.504227
        207982 |         293757 | batch4           |        20 | lot       | 2019-03-11 16:06:50.994719 | 2019-03-11 16:08:24.823359
        207979 |         293757 | batch1           |        20 | lot       | 2019-03-11 16:05:10.249634 | 2019-03-11 16:06:43.364219
        208001 |         293757 | batch1           |        20 | lot       | 2019-03-11 16:08:31.927953 | 2019-03-11 16:10:05.1156
        208005 |         293757 | batch5           |        20 | lot       | 2019-03-11 16:10:12.654001 | 2019-03-11 16:11:46.213178
```

**Comment by mew2057** [comment]
I have a candidate fix, do we want this on 1.6.0 or 1.5.

**Comment by fpizzano** [comment]
Adding interested parties: @miidon and @mattaezell

**Comment by fpizzano** [comment]
@awlauria and I tested ssh and csm launch with latest JSM & CSM build and all work:

```
$ rpm -qa | grep ibm-csm
ibm-csm-hcdiag-1.7.0-2735.noarch
ibm-csm-core-1.7.0-2735.ppc64le
ibm-csm-api-1.7.0-2735.ppc64le
$ rpm -qa | grep smpi
ibm_smpi-10.03.00.01rtm3-rh7_20190611.ppc64le
ibm_smpi-devel-10.03.00.01rtm3-rh7_20190611.ppc64le
ibm_smpi_lic_s-10.03.01rtm3-rh7_20190611.ppc64le
ibm_smpi-jsm-10.03.00.01rtm3-rh7_20190611.ppc64le
$
```

```
$ ./jslist
     parent                cpus      gpus      exit
   ID   ID       nrs    per RS    per RS    status         status
===============================================================================
    1    0        12         1         0         0       Complete
    2    0        12         1         0         0       Complete
    3    0        12         1         0         0       Complete
```

**Comment by fpizzano** [comment]
New behavior with:

```
# head -1 /var/log/ibm/csm/csm_utility.log
[UTL]2019-07-25 14:28:57.782804       csmd::info     | CSMD: UTILITY; Version: 1.7.0; Build: a29b8a31e3; Date: Wed Jul 24 15:34:53 2019 -0400; Config file: /etc/ibm/csm/csm_utility.cfg
#
```

```
# /opt/ibm/csm/bin/csm_allocation_query -j 101
---
num_allocations: 4
allocations:
  - allocation_id:    2
    primary_job_id:   101
    secondary_job_id: 0
    begin_time:      2019-07-22 06:53:29.428893
    state:           complete
    history:
     end_time:     2019-07-22 06:54:10.891832
     exit_status:  0
  - allocation_id:    3
    primary_job_id:   101
    secondary_job_id: 0
    begin_time:      2019-07-22 06:54:28.922257
    state:           complete
    history:
     end_time:     2019-07-23 01:49:24.904938
     exit_status:  0
  - allocation_id:    4
    primary_job_id:   101
    secondary_job_id: 0
    begin_time:      2019-07-23 01:49:43.39492
    state:           complete
    history:
     end_time:     2019-07-23 02:25:54.807362
     exit_status:  0
  - allocation_id:    5
    primary_job_id:   101
    secondary_job_id: 0
    begin_time:      2019-07-23 02:25:58.836329
    state:           running
...
```

**Comment by besawn** [comment]
Issue has been fixed since CSM 1.7.0.
